### PR TITLE
[PoC] Spec theme switcher

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=IE4eaDN)
#### Summary

This is just a proof of concept to test the problems we are facing with theming support and Percy testing.

### CODE REVIEW

File: design-system/src/theme-provider.tsx [hunk 1]
```diff
@@ -17,7 +17,15 @@ type ThemeName = keyof typeof themes;
...
 
-const ThemeContext = createContext({});
+type TThemeContext = {
+  theme: ThemeName;
+  changeTheme: (newTheme: string) => void;
+};
+
+const ThemeContext = createContext<TThemeContext>({
+  theme: themesNames.default,
+  changeTheme: () => {},
```
Comment (kark) [inline design-system/src/theme-provider.tsx:27]: Thanks!

File: design-system/src/theme-provider.visualroute.jsx [hunk 2]
```diff
@@ -1,34 +1,42 @@
...
-  const { changeTheme } = useTheme();
-  useLayoutEffect(() => {
-    changeTheme('dark');
-  }, [changeTheme]);
-
-  return (<DummyComponent />);
-};
+    }}
+  >
+    Title with default theme <i>colorPrimary</i> design token
```
Comment (kark) [inline design-system/src/theme-provider.visualroute.jsx:12]: The only issue I see is that `default theme` is hardcoded here and rendered in all cases. If you don't feel like changing that here I can take care of that later.

**Review by kark — APPROVED** [review]
Work great 🙌

File: design-system/src/theme-provider.visualroute.jsx (hunk 2 above)
Comment (CarlosCortizasCT) [reply design-system/src/theme-provider.visualroute.jsx]: Thanks for pointing that out 👍 

Updated in this commit: 9707ed017093bcaf474fe321006de4d3139725b7

**Review by kark — APPROVED** [review]

**Pull request merged**